src/sql/reuse.rs

```diff
@@ -320,4 +362,173 @@ mod tests {
         assert!(!predicate_implies(&pa, &pc));
         assert!(predicate_implies(&pc, &pa));
     }
+
+    #[test]
+    fn complex_predicate_implication_or() {
+        use nom_sql::ConditionExpression::*;
+        use nom_sql::ConditionBase::*;
+        use nom_sql::Literal;
+
+        let pa = ComparisonOp(ConditionTree {
+            operator: Operator::Less,
+            left: Box::new(Base(Field(Column::from("a")))),
+            right: Box::new(Base(Literal(Literal::Integer(20.into())))),
+        });
+        let pb = ComparisonOp(ConditionTree {
+            operator: Operator::Greater,
+            left: Box::new(Base(Field(Column::from("a")))),
+            right: Box::new(Base(Literal(Literal::Integer(60.into())))),
+        });
+        let pc = ComparisonOp(ConditionTree {
+            operator: Operator::Less,
+            left: Box::new(Base(Field(Column::from("a")))),
+            right: Box::new(Base(Literal(Literal::Integer(10.into())))),
+        });
+        let pd = ComparisonOp(ConditionTree {
+            operator: Operator::Greater,
+            left: Box::new(Base(Field(Column::from("a")))),
+            right: Box::new(Base(Literal(Literal::Integer(80.into())))),
+        });
+
+        // a < 20 or a > 80
+        let cp1 = LogicalOp(ConditionTree {
+            left: Box::new(pa.clone()),
+            right: Box::new(pb.clone()),
+            operator: Operator::Or
+        });
+
+        // a < 10 or a > 60
+        let cp2 = LogicalOp(ConditionTree {
+            left: Box::new(pc),
+            right: Box::new(pd),
+            operator: Operator::Or
+        });
+
+        // a < 80 or a > 20
+        let cp3 = LogicalOp(ConditionTree {
+            left: Box::new(pb),
+            right: Box::new(pa),
+            operator: Operator::Or
+        });
```

Comment:
Is it deliberate that this predicate is tautological? (There is no `a` that is greater than 80 and less than 20.) The implications are correct, I was just curious if this is a deliberate choice.